processor/metricstransformprocessor/config.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import "go.opentelemetry.io/collector/config/configmodels"
+
+// Config defines configuration for Resource processor.
+type Config struct {
+	configmodels.ProcessorSettings `mapstructure:",squash"`
+
+	// MetricName is used to select the metric to operate on.
+	// REQUIRED
+	MetricName string `mapstructure:"metric_name"`
+
+	// Action specifies the action performed on the matched metric.
+	// REQUIRED
+	Action ConfigAction `mapstructure:"action"`
+
+	// NewName specifies the name of the new metric when inserting or updating.
+	// REQUIRED only if Action is INSERT.
+	NewName string `mapstructure:"new_name"`
```

Comment:
I think the optional field should be called SourceMetricName, so MetricName should always be the output metric name, and the input metric should default to the same as the output metric.

Reply:
So basically, do you mean an optional `SourceMetricName` field, and a required `MetricName` field? When both are set, it is a rename from `SourceMetricName` to `MetricName`. In this case, wouldn't the user be confused about if the `MetricName` means the current name or the new name because `MetricName` would represent two different meanings given different contexts.